runtime/shared_heap.c

```diff
@@ -47,6 +47,7 @@ typedef struct pool {
   value* next_obj;
   caml_domain_state* owner;
   sizeclass sz;
+  long p1; long p2; long p3; long p4; /* padding to align slabs */
```

Comment:
I'm no C expert, but my impression is that this code is making assumptions about the word size of `long` that may not hold under all support configurations (at least if we include bytecode-only targets). I suspect that there is a caml-runtime-specific type alias that is more appropriate, maybe just `uintnat`. But should we also redefine `sizeclass` to be `uintnat` if we want to be able to make such padding assumptions?

Also, should this be `uintnat padding[4]` instead?

Finally, does this padding-to-align logic also work on systems with 32bit machine words? I think that you are asserting that the first element in a pool is 64-bytes-aligned, and assuming that a 8-word headers ensures that, which may not hold on 32bit systems.